src/components/Proposal/Stakes/index.tsx

```diff
@@ -0,0 +1,329 @@
+import { useState } from 'react';
+import moment from 'moment';
+import { useLocation } from 'react-router-dom';
+
+import { useContext } from 'contexts';
+import {
+  bnum,
+  calculateStakes,
+  formatBalance,
+  denormalizeBalance,
+} from 'utils';
+
+import { FiThumbsUp, FiThumbsDown } from 'react-icons/fi';
+
+import {
+  Question,
+  AmountBadge,
+  BlockchainLink,
+  HorizontalSeparator,
+} from 'components/common';
+
+import {
+  ActionArea,
+  SidebarRow,
+  Vote,
+  AmountInput,
+  PositiveSummary,
+  NegativeSummary,
+  SummaryDetails,
+  SummaryTotal,
+  ActionButton,
+} from '../styles';
+
+const Stakes = () => {
+  const {
+    context: { daoStore, configStore, providerStore, daoService, userStore },
+  } = useContext();
+
+  const [stakeAmount, setStakeAmount] = useState(0);
+  // We should get the ID in another way
+  const proposalId = useLocation().pathname.split('/')[3];
+  console.log('Proposal ID in Vote: ', proposalId);
+  const proposal = daoStore.getProposal(proposalId);
+  const proposalEvents = daoStore.getProposalEvents(proposalId);
+  const { account } = providerStore.getActiveWeb3React();
+  const scheme = daoStore.getScheme(proposal.scheme);
+
+  let stakedAmount = bnum(0);
+  let positiveStakesCount = proposalEvents.stakes.filter(
+    stake => stake.vote.toString() === '1'
+  ).length;
+  let negativeStakesCount = proposalEvents.stakes.filter(
+    stake => stake.vote.toString() === '2'
+  ).length;
+  const networkContracts = configStore.getNetworkContracts();
+  const votingMachines = networkContracts.votingMachines;
+
+  const redeemsLeft = daoStore.getUserRedeemsLeft(account);
+
+  const votingMachineTokenName =
+    votingMachines.gen && scheme.votingMachine === votingMachines.gen.address
+      ? 'GEN'
+      : 'DX';
+
+  const { dxdApproved, genApproved } = userStore.getUserInfo();
+  const votingMachineTokenApproved =
+    votingMachines.gen && scheme.votingMachine === votingMachines.gen.address
+      ? genApproved
+      : dxdApproved;
+  const votingParameters = daoStore.getVotingParametersOfProposal(proposalId);
+
+  proposalEvents.stakes.map(stake => {
+    if (stake.staker === account && stake.vote.toString() === '1') {
+      stakedAmount = stakedAmount.plus(stake.amount);
+    } else if (stake.staker === account && stake.vote.toString() === '2') {
+      stakedAmount = stakedAmount.minus(stake.amount);
+    }
+  });
+
+  const { recommendedStakeToBoost, recommendedStakeToUnBoost } =
+    calculateStakes(
+      votingParameters.thresholdConst,
+      scheme.boostedProposals,
+      proposal.stateInVotingMachine === 4
```

Comment:
I think we should stop comparing enums to numbers as its not clear what is happening here. thoughts?

Reply:
Yes, we should. Lets leave it for a future PR though,